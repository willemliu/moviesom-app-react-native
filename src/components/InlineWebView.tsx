import React from 'react';
import { webViewStyle, transparentColor } from '../styles/Styles';
import {WebView, ActivityIndicator, View, TextInput, BackHandler, TextStyle, StyleProp} from 'react-native';

export interface Props {
    url: string;
    style?: StyleProp<TextStyle>;
    hideAddressBar?: boolean;
    canGoBack?: boolean;
}

export default class InlineWebView extends React.Component<Props, any> {
    state: any = {
        loading: false
    };

    private webRef: any;
    private lastUrl: any;

    constructor(props: any) {
        super(props);
    }

    componentDidMount() {
        if (this.props.canGoBack) {
            BackHandler.addEventListener('hardwareBackPress', this.backHandler);
        }
    }

    backHandler = (): boolean => {
        if (this.webRef && this.state.url !== this.lastUrl && this.props.canGoBack && this.state.canGoBack) {
            this.lastUrl = this.state.url;
            this.webRef.goBack();
            return true;
        }
        return false;
    }

    componentWillUnmount() {
        BackHandler.removeEventListener('hardwareBackPress', this.backHandler);
    }

    onNavigationStateChange = (navState: any) => {
        this.setState({
            ...navState,
        });
    }

    render() {
        return (
            <View style={{flex: 1}}>
                {this.props.hideAddressBar ? null :
                <TextInput
                    selectTextOnFocus={true}
                    underlineColorAndroid={transparentColor}
                    multiline={true}
                    style={{
                        textAlign: 'center',
                        padding: 5,
                }}>{this.props.url}</TextInput>}
                <WebView
                    ref={(ref: any) => this.webRef = ref}
                    source={{uri: this.props.url}}
                    startInLoadingState={false}
                    onNavigationStateChange={this.onNavigationStateChange}
                    style={this.props.style ? this.props.style : webViewStyle.webView}
                />
                {this.state.loading ? <ActivityIndicator size='large' color='#009688' style={webViewStyle.activityIndicator}/> : null}
            </View>
        );
    }
}
