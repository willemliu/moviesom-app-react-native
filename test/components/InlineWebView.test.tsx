import React from 'react';
import renderer from 'react-test-renderer';
import InlineWebView from "../../src/components/InlineWebView";

describe('InlineWebView', () => {
    it('should render correctly', () => {
        const rendered = renderer.create(<InlineWebView url="https://www.google.com" />).toJSON();
        expect(rendered).toMatchSnapshot();
    });
});
