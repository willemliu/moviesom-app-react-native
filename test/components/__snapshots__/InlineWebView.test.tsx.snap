// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineWebView should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    multiline={true}
    selectTextOnFocus={true}
    style={
      Object {
        "padding": 5,
        "textAlign": "center",
      }
    }
    underlineColorAndroid="transparent"
  >
    https://www.google.com
  </TextInput>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTWebView
      allowUniversalAccessFromFileURLs={undefined}
      automaticallyAdjustContentInsets={undefined}
      contentInset={undefined}
      domStorageEnabled={undefined}
      injectedJavaScript={undefined}
      javaScriptEnabled={true}
      mediaPlaybackRequiresUserAction={undefined}
      messagingEnabled={false}
      mixedContentMode={undefined}
      onContentSizeChange={undefined}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      saveFormDataDisabled={false}
      scalesPageToFit={true}
      source={
        Object {
          "uri": "https://www.google.com",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "marginTop": 0,
          },
        ]
      }
      testID={undefined}
      thirdPartyCookiesEnabled={true}
      urlPrefixesForDefaultIntent={undefined}
      userAgent={undefined}
    />
  </View>
</View>
`;
